src/Tools/CoreFx.Tools/FindBestConfiguration.cs

```diff
@@ -0,0 +1,119 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class FindBestConfiguration : BuildTask
+    {
+        private static readonly char s_SupportedGroupsSeparator = '|';
+        private static readonly char s_VerticalGroupsSeparator = '-';
```

Comment:
Don't pay attention to the task.  I just copied @chcosta's to show how this would work.  I don't think he was ready for review yet.